Source/JavaScriptCore/runtime/StringPrototype.cpp

```diff
@@ -934,30 +934,19 @@ JSC_DEFINE_HOST_FUNCTION(stringProtoFuncSubstring, (JSGlobalObject* globalObject
     int len = jsString->length();
     RELEASE_ASSERT(len >= 0);
 
-    double start = a0.toNumber(globalObject);
+    double start = a0.toIntegerOrInfinity(globalObject);
     RETURN_IF_EXCEPTION(scope, { });
+    start = std::clamp<double>(start, 0.0, len);
```

Comment:
We would probably get slightly better performance by doing fewer conversions from `double` to `unsigned` if we did this:
```suggestion
    double startDouble = a0.toIntegerOrInfinity(globalObject);
    RETURN_IF_EXCEPTION(scope, { });
    unsigned start = std::clamp<double>(startDouble, 0.0, len);
```

Reply:
Fixed